packages/desktop-client/src/components/FinancesApp.tsx

```diff
@@ -88,20 +89,53 @@ function RouterBehaviors() {
 
 export function FinancesApp() {
   const dispatch = useDispatch();
+  const { t } = useTranslation();
+
+  const [lastUsedVersion, setLastUsedVersion] = useLocalPref(
+    'flags.updateNotificationShownForVersion',
+  );
+
   useEffect(() => {
     // Wait a little bit to make sure the sync button will get the
     // sync start event. This can be improved later.
     setTimeout(async () => {
       await dispatch(sync());
-
-      await checkForUpdateNotification(
-        dispatch,
-        getIsOutdated,
-        getLatestVersion,
-      );
```

Comment:
Inlined the update checking logic. This makes it easier to use various other hooks in it (i.e. `useLocalPref` and `t`).